### PR TITLE
feat: integrate widget tx states

### PR DESCRIPTION
**Pull request by JFrankfurt**
image

### CODE REVIEW

File: src/components/Widget/transactions.ts [hunk 1]
```diff
@@ -1,17 +1,76 @@
...
+    [addTransaction, allowedSlippage, chainId]
+  )
+
   const txHandlers: TransactionEventHandlers = useMemo(
     () => ({
-      onTxSubmit: (hash: string, tx: unknown) => console.log('onTxSubmit'),
-      onTxSuccess: (hash: string, receipt: TransactionReceipt) => console.log('onTxSuccess'),
-      onTxFail: (hash: string, receipt: TransactionReceipt) => console.log('onTxFail'),
+      onTxSubmit,
+      onTxSuccess: (hash: string, receipt: TransactionReceipt) => console.log('onTxSuccess', hash, receipt),
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:70]: If they're unneeded, rm `onTxSuccess`/`onTxFail` handlers

File: src/components/Widget/transactions.ts [hunk 2]
```diff
@@ -1,17 +1,76 @@
...
+
+      if (!type || !response) {
+        return
+      }
+      if (type === TransactionType.WRAP || type === TransactionType.UNWRAP) {
+        const { amount } = transaction.info
+
+        addTransaction(response, {
+          type: TransactionType.WRAP,
+          unwrapped: false,
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:34]: This should be dependent on `TransactionType`, maybe:

    unwrapped: type === TransactionType.UNWRAP

File: src/components/Widget/transactions.ts [hunk 3]
```diff
@@ -1,17 +1,76 @@
...
+      const { type, response } = transaction.info
+
+      if (!type || !response) {
+        return
+      }
+      if (type === TransactionType.WRAP || type === TransactionType.UNWRAP) {
+        const { amount } = transaction.info
+
+        addTransaction(response, {
+          type: TransactionType.WRAP,
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:33]: This must be the app's TransactionType

    import { TransactionType as AppTransactionType } from 'state/transactions/types'

Then you can drop the `unknown` conversion.

File: src/components/Widget/transactions.ts [hunk 4]
```diff
@@ -1,17 +1,76 @@
...
+          unwrapped: false,
+          currencyAmountRaw: amount.quotient.toString(),
+          chainId,
+        } as unknown as WrapTransactionInfo)
+      }
+      if (type === TransactionType.SWAP) {
+        const { trade, tradeType } = transaction.info
+        const baseTxInfo = {
+          type: TransactionType.SWAP,
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:42]: This must be the app's TransactionType

    import { TransactionType as AppTransactionType } from 'state/transactions/types'

Then you can drop the `unknown` conversion.

File: src/components/Widget/transactions.ts [hunk 5]
```diff
@@ -1,17 +1,76 @@
...
+import { useUserSlippageTolerance } from 'state/user/hooks'
+import { currencyId } from 'utils/currencyId'
 
 /** Integrates the Widget's transactions, showing the widget's transactions in the app. */
 export function useSyncWidgetTransactions() {
-  // TODO(jfrankfurt): Integrate widget transactions with app transaction tracking.
+  const { chainId } = useWeb3React()
+  const addTransaction = useTransactionAdder()
+  const [appSlippage] = useUserSlippageTolerance()
+  const allowedSlippage = appSlippage === 'auto' ? V3_SWAP_DEFAULT_SLIPPAGE : appSlippage
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:20]: The slippage is actually defined at the time of the transaction by the widget, and the default may not match the interface default. To keep the definition in one reliable place, I think it's best to include this in the trade object emitted by the widget (which will need to be added to it).

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/Widget/transactions.ts (hunk 5 above)
Comment (JFrankfurt) [reply src/components/Widget/transactions.ts:20]: agree--this default value is what I felt most uncomfortable about in the pr

File: src/components/Widget/transactions.ts [hunk 6]
```diff
@@ -26,20 +32,20 @@ export function useSyncWidgetTransactions() {
...
-          unwrapped: false,
+          type: AppTransactionType.WRAP,
+          unwrapped: type === WidgetTransactionType.UNWRAP,
           currencyAmountRaw: amount.quotient.toString(),
           chainId,
-        } as unknown as WrapTransactionInfo)
+        } as WrapTransactionInfo)
       }
-      if (type === TransactionType.SWAP) {
+      if (type === WidgetTransactionType.SWAP) {
```
Comment (zzmp) [inline src/components/Widget/transactions.ts:45]: nit: `else if`

**Review by zzmp — CHANGES_REQUESTED** [review]

File: src/components/Widget/transactions.ts (hunk 5 above)
Comment (zzmp) [reply src/components/Widget/transactions.ts:26]: See https://github.com/Uniswap/widgets/pull/187

Comment (JFrankfurt) [reply src/components/Widget/transactions.ts]: done

**Review by zzmp — APPROVED** [review]

**Pull request merged**